python_modules/dagster/dagster/_core/definitions/composition.py

```diff
@@ -64,6 +64,91 @@ class MappedInputPlaceholder:
     """Marker for holding places in fan-in lists where input mappings will feed."""
 
 
+class InvokedNodeOutputHandle:
+    """The return value for an output when invoking a node in a composition function."""
+
+    solid_name: str
+    output_name: str
+    node_type: str
+
+    def __init__(self, solid_name: str, output_name: str, node_type: str):
+        self.solid_name = check.str_param(solid_name, "solid_name")
+        self.output_name = check.str_param(output_name, "output_name")
+        self.node_type = check.str_param(node_type, "node_type")
```

Comment:
nit: this is internally constructed only right? could drop `check` calls and do condensed namedtuple or similar

Reply:
like the other entries in the source union